tinytuya/Contrib/ThermostatDevice.py

```diff
@@ -273,6 +273,7 @@ def __init__(self, *args, **kwargs):
         self.delayed_updates = { }
         self.sensorlists = [ ]
         self.sensors = self.SensorList( self )
+        self.mode = None

```

Comment:
Is this to make pylint happy?  It won't hurt anything, but it's not really needed as it gets set via setattr() in the for loop below.

Reply:
Great point and I almost dismissed it myself. At first I thought it might be a miss on our part, but then I saw it would work once setattr() is called in the flow which is why it wouldn't error.  I believe the basis for initializing all the classed based variables in __init__() is to help with readability.  I was confused at first before I traced it myself. Having the class scoped variables all listed in init() seems clean (maybe a bit pedantic too? :).

Reply:
I hate duplicating stuff though, and listing everything would require duplicating the entire `dps_data` dict.  I just find it cleaner to have everything defined in 1 place instead of having smaller bits strewn about; this way there's no forgetting to initialize it in some other place after defining it here.

Reply:
Fair. It's a style issue and not functional. Pylint (and vscode) are not able to unwind the setattr() to see that `self.mode` is actually defined in init().  but as you say, changing it would likely double the confusion and prevent someone from unwinding how you are doing the setter. I'll revert.